hazelcast/src/main/java/com/hazelcast/client/UndefinedErrorCodeException.java

```diff
@@ -20,8 +20,16 @@
 import com.hazelcast.spi.impl.operationservice.WrappableException;
 
 /**
- * This exception is thrown when an exception that is coming from server is not recognized by the protocol.
- * Class name of the original exception is included in the exception
+ * This exception is thrown when an exception that is coming from server is not recognized by the protocol and
+ * it can not be constructed by the client via reflection.
+ * For the client to be able to recreate original exception it should be available on the classpath and
+ * it should have one of the following constructors publicly.
+ * new Throwable(String message, Throwable cause)
```

Comment:
nit: a list (ul/li) + code blocks could be used here to make this part of the javadoc more readable